packages/vats/src/proposals/upgrade-vats-generic-proposal.js

```diff
@@ -49,3 +140,47 @@ export const getManifestForUpgradingVats = (
   },
   options: { bundleRefs, vatOptions },
 });
+
+/**
+ * @param {any} powers
+ * @param {{
+ *   contractKitSpecs: ContractKitSpec[];
+ *   installationBundleRefs: { [bundleName: string]: VatSourceRef };
+ * }} options
+ * @returns
+ */
+export const getManifestForUpgradingZoeContractKits = (
+  { restoreRef },
+  { contractKitSpecs, installationBundleRefs },
+) => {
+  const consumePermits = {
+    instancePrivateArgs: true,
+  };
+  for (const { kitLookup } of contractKitSpecs) {
+    let current = consumePermits;
+    for (let i = 0; i < kitLookup.length - 1; i += 1) {
+      current = current[kitLookup[i]] || (current[kitLookup[i]] = {});
+    }
+    current[kitLookup[kitLookup.length - 1]] ||= true;
+  }
+
+  return {
+    manifest: {
+      [upgradeZoeContractsGeneric.name]: {
+        // Other consume permits should only be added in the definition of
+        // `consumePermits`, at the top of this function.
+        consume: consumePermits,
+      },
+    },
+    installations: Object.fromEntries(
+      Object.entries(installationBundleRefs).map(([bundleName, bundleRef]) => [
+        bundleName,
+        restoreRef(bundleRef),
+      ]),
+    ),
```

Comment:
don't we get that for free?

I'm not confident the "for free" mechanism is reliable, though. My core-evals seem to get old installations sometimes. Is this intended to address that?